src/ucs/memory/memtype_cache.c

```diff
@@ -29,8 +29,13 @@ typedef enum {
 
 static ucs_pgt_dir_t *ucs_memtype_cache_pgt_dir_alloc(const ucs_pgtable_t *pgtable)
 {
-    return ucs_memalign(UCS_PGT_ENTRY_MIN_ALIGN, sizeof(ucs_pgt_dir_t),
-                        "memtype_cache_pgdir");
+    void *ptr;
+    int ret;
+
+    ret = ucs_posix_memalign(&ptr,
+                             ucs_max(sizeof(void *), UCS_PGT_ENTRY_MIN_ALIGN),
```

Comment:
as we talked, need to make sure the macro UCS_PGT_ENTRY_MIN_ALIGN is already defined to be at least sizeof(void*) instead of doing ucs_max every time it's used

Reply:
Are you fine with
`#define UCS_PGT_ENTRY_MIN_ALIGN (ucs_max(sizeof(void *), UCS_PGT_ENTRY_FLAGS_MASK + 1)` ?
I do not think it is appropriate to make assumptions about the pointer size.

Reply:
ok, this is a good point, so let's keep this as is.
so just need to fix indentation in several posix_memalign usages (indent on column)